src/transaction/log_recovery.c

```diff
@@ -2145,9 +2268,12 @@ log_recovery_abort_atomic_sysop (THREAD_ENTRY * thread_p, LOG_TDES * tdes)
  *
  * return: nothing
  *
+ * skip_some_page_buffer_flushes (in) : skip flushes to log page buffer header (mainly needed when
+ *               function called in the context of recovering a transaction server with
+ *               remote storage)
  */
 static void
-log_recovery_undo (THREAD_ENTRY * thread_p)
+log_recovery_undo (THREAD_ENTRY * thread_p, bool skip_some_page_buffer_flushes)
```

Comment:
Since there are only 2 calls. It would be better to take all the flush* functions out of undo function.